gio/src/task.rs

```diff
@@ -433,10 +433,12 @@ where
     // use Cancellable::NONE as source obj to fulfill `Send` requirement
     let task = unsafe { Task::<bool>::new(Cancellable::NONE, Cancellable::NONE, |_, _| {}) };
     let (join, tx) = JoinHandle::new();
-    task.run_in_thread(move |_, _: Option<&Cancellable>, _| {
-        let res = panic::catch_unwind(panic::AssertUnwindSafe(func));
-        let _ = tx.send(res);
-    });
+    task.clone()
+        .run_in_thread(move |_, _: Option<&Cancellable>, _| {
```

Comment:
```suggestion
    task.run_in_thread(move |task, _: Option<&Cancellable>, _| {
```

seems better or am I missing something?